### PR TITLE
Dpev/streaming textures

### PR DESCRIPTION
**Pull request by dpev**
Added streaming texture interface, plus a few other quality-of-life fixes

### CODE REVIEW

File: Source/udImage.cpp [hunk 1]
```diff
@@ -121,12 +141,291 @@ void udImage_Destroy(udImage **ppImage)
...
+  UD_ERROR_CHECK(udFile_Read(pFile, pImage, sizeof(udImageStreamingOnDisk), offset, udFSW_SeekSet));
+  UD_ERROR_IF(pImage->fourcc != udImageStreamingOnDisk::Fourcc, udR_ObjectTypeMismatch);
+  UD_ERROR_IF(pImage->mipCount > udImageStreamingOnDisk::MaxMipLevels, udR_CorruptData);
+  pImage->pFile = pFile;
+  pImage->baseOffset = offset;
+  pImage->pLock = udCreateMutex();
+
+  for (uint16_t mip = 0; mip < pImage->mipCount; ++mip)
+  {
+    pImage->mips[mip].offset = (!mip) ? offset + pImage->offsetToMip0 : pImage->mips[mip - 1].offset + (pImage->mips[mip - 1].width * pImage->mips[mip - 1].height * 3);
```
Comment (samsinsane) [inline Source/udImage.cpp:285]: Not a huge fan of the complexity and lack of brackets here.

Comment (dpev) [reply Source/udImage.cpp:285]: Yeah agree actually, wanna hang 5 and I'll quickly fix?

Comment (samsinsane) [reply Source/udImage.cpp:285]: Sounds good.

Comment (dpev) [reply Source/udImage.cpp]: Fixed now

**Pull request merged**